tests/console/zypper_lifecycle.pm

```diff
@@ -77,8 +77,9 @@ sub run {
     assert_script_run "rm -f /var/lib/lifecycle/data/$prod.lifecycle";
 
     # get product eol
-    my $product_name = script_output "grep '<summary>' /etc/products.d/$prod.prod";
-    $product_name =~ s/.*<summary>([^<]*)<\/summary>.*/$1/;
+    my $product_file = "/etc/products.d/$prod.prod";
+    my $product_name = script_output "grep '<summary>' $product_file";
+    $product_name =~ s/.*<summary>([^<]*)<\/summary>.*/$1/ || die "no product name found in $product_file";
```

Comment:
You could add the actual value of $product_name into the message. This way the reviewer could check the screenshots if this string is really not there or if it (while doing grep) just found something wrong in product_file.

Something like "no product called \"$product_name\" found in $product_file" could be useful.

Reply:
but product_name would contain "<summary>" then...